src/sentry/api/bases/organization_events.py

```diff
@@ -525,6 +525,10 @@ def get_event_stats_data(
                             zerofill_results=zerofill_results,
                             dataset=dataset,
                         )
+                        if request.query_params.get("useOnDemandMetrics") == "true":
```

Comment:
If top n events are called but we are not using on-demand, there's no reason to even try the code.